types/rc-form/index.d.ts

```diff
@@ -0,0 +1,149 @@
+        callback: ValidateCallback<TFields>,
+    ): void;
+    validateFields(options: ValidateFieldsOptions, callback: ValidateCallback<TFields>): void;
+    validateFields(
+        fieldNames: string[],
+        callbackOrOptions: ValidateCallback<TFields> | ValidateFieldsOptions
+    ): void;
+    validateFields(callbackOrOptions: ValidateCallback<TFields> | ValidateFieldsOptions): void;
+    validateFields(fieldNames?: string[]): void;
+    getFieldError(name: string): object[];
+    getFieldsError(names?: string[]): object;
+    isFieldValidating(name: string): boolean;
+    isFieldTouched(name: string): boolean;
+    isFieldsTouched(names?: string[]): boolean;
+    resetFields(names?: string[]): void;
+    getFieldDecorator(
+        id: string,
+        options?: GetFieldDecoratorOptions,
+    ): (node: React.ReactNode) => React.ReactNode;
+}
+
+export interface RcBaseFormProps {
+    wrappedComponentRef?: any;
+}
+
+export interface WrappedFormInternalProps<V = any> {
+    form: WrappedFormUtils<V>;
+}
+
+export interface FormComponentProps<V = any> extends WrappedFormInternalProps<V>, RcBaseFormProps {
+    form: WrappedFormUtils<V>;
+}
+
+type FormCreateOptionMessagesCallback = (...args: any[]) => string;
+
+interface FormCreateOptionMessages {
+[messageId: string]: string | FormCreateOptionMessagesCallback | FormCreateOptionMessages;
+}
+
+interface Field<T = any> {
+    name?: string;
+    value?: T;
+    errors?: string[];
+    dirty?: boolean;
+    touched?: boolean;
+}
+
+type Primitive = null | undefined | number | string | symbol;
+
+type Fields<T> = {
+    [K in keyof T]: T[K] extends Primitive
+        ? Field<T[K]>
+        : T[K] extends any[]
+            ? { [Key: number]: Fields<T[K][number]>}
+            : Fields<T[K]>
+};
+
+export interface FormCreateOption<TProps extends FormComponentProps> {
+    onFieldsChange?: (
+    props: TProps,
+    changedFields: Fields<GetFields<TProps>>,
+    allFields: Fields<GetFields<TProps>>,
+    ) => void;
+    onValuesChange?: (props: TProps, changedValues: Fields<GetFields<TProps>>, allValues: Fields<GetFields<TProps>>) => void;
+    mapProps?: (props: TProps) => WrappedFormUtils<GetFields<TProps>>;
+    mapPropsToFields?: (props: TProps) => Fields<GetFields<TProps>>;
+    validateMessages?: FormCreateOptionMessages;
+    withRef?: boolean;
+    fieldNameProp?: string;
+    fieldMetaProp?: string;
+    fieldDataProp?: string;
+}
+
+type FormWrappedProps<TOwnProps extends WrappedFormInternalProps> = (
+    component: React.ComponentType<TOwnProps>
+) => React.ComponentType<TOwnProps>;
+
+interface ComponentDecorator<P> {
+(
+    component: React.ComponentClass<P> | React.SFC<P>,
+): React.ComponentClass<RcBaseFormProps & Omit<P, keyof FormComponentProps>>;
+}
+
+type GetFields<T> = T extends FormComponentProps<infer A> ? A : {};
+
+export function createForm<TOwnProps extends FormComponentProps>(options?: FormCreateOption<TOwnProps>): ComponentDecorator<TOwnProps>;
+
+export {};
```

Comment:
is this needed? Usually it's just a trick to turn a non-module into a module, but there are lots of other exports that turn this into a module already.